scripts/skills/effects/rf_encumbrance_effect.nut

```diff
@@ -130,26 +130,46 @@ this.rf_encumbrance_effect <- ::inherit("scripts/skills/skill", {
 		switch (this.getEncumbranceLevel())
 		{
 			case 0:
-				return;
-
 			case 1:
-				_properties.FatigueRecoveryRate -= 1;
-				break;
+				return;
 
 			case 2:
-				_properties.FatigueRecoveryRate -= 2;
 				_properties.MovementFatigueCostAdditional += 1;
 				break;
 
 			case 3:
-				_properties.FatigueRecoveryRate -= 2;
 				_properties.MovementFatigueCostAdditional += 3;
 				break;
 
 			default:
-				_properties.FatigueRecoveryRate -= 3;
 				_properties.MovementFatigueCostAdditional += 3;
 				break;
 		}
 	}
+
+	function onTurnStart()
+	{
+		local fatigue;
+		switch (this.getEncumbranceLevel())
+		{
+			case 0:
+				return;
+
+			case 1:
+				fatigue = 1;
+				break;
+
+			case 2:
+			case 3:
+				fatigue = 2;
+				break;
+
+			default:
+				fatigue = 3;
+				break;
+		}
+
+		local actor = this.getContainer().getActor();
+		actor.setFatigue(::Math.min(actor.getFatigueMax(), actor.getFatigue() + fatigue));
```

Comment:
This would fail on brothers with less than 15 usable fatigue.
A brother with 10/10 Fatigue would recover 10 Fatigue due to vanilla mechanic leaving him at 0/10 Fatigue at the start of their turn.
If they have the highest Encumbrance Level he now gains a penalty of 3 fatigue leaving him at 3/10 Fatigue.

What you actually want is him being at 10/10 Fatigue because you reduce the 15 Fatigue per turn down to 12 Fatigue per turn.

Reply:
Lol yeah I realized this when I was going to sleep last night. It will fail also when someone recovers enough Fatigue to get to 0 but had excess fatigue recovery. E.g. a bro has 10 fatigue at turn start and has level 2 encumbrance. Due to fatigue recovery he should still go down to 0 fatigue but encumbrance will add fatigue to him.

So yeah this solution doesn't work. The only solution I can think of is to hook the actor.nut function which is unfortunate.

So instead of doing that perhaps encumbrance should increase fatigue cost of skills instead of affecting fatigue recovery? The goal is to eliminate the fatigue neutral cheese builds. As long as we achieve that goal in a sensible way it should be fine.

Reply:
@Darxo , @Uberbagel If we want a clean and compatible solution, it would be to change the Encumbrance mechanic to say "When encumbered you build x Fatigue at the start of every turn". It accomplishes the goal of removing fatigue neutral cheese builds, and is then consistent with all edge cases.